cmd/common/config.go

```diff
@@ -52,11 +52,21 @@ func SetFlagsForParams(cmd *cobra.Command, params *params.Params, configPrefix s
 		if f == nil {
 			continue
 		}
-		// Apply the config value to the flag when the flag is not set and config value is set
-		if !f.Changed && config.Config.IsSet(configPrefix+k) {
-			err := f.Value.Set(config.Config.GetString(configPrefix + k))
-			if err != nil {
-				return fmt.Errorf("setting flag %s: %w", k, err)
+
+		// Apply the config value to the flag when the flag is not set and env/config value is set
+		if !f.Changed {
+			// we need to bind env without the prefix to avoid long names like INSPEKTOR_GADGET_OPERATOR_NAME_KEY
+			config.Config.BindEnv(k)
+
+			// if env is set, use it otherwise use the config value
```

Comment:
I think it's fine to have a 1 to 1 mapping between env variable and cli flag. I agree the name clash issue is also present for the cli flags, so I'd prefer to continue without blocking on that. 

/cc @flyth

Reply:
Operators and runtimes use a `Key` when defining a parameter, this key is [used](https://github.com/inspektor-gadget/inspektor-gadget/blob/731603ac82260a24c427cee836944ffc269c3a24/cmd/common/registry.go#L750) as the CLI flag name:

https://github.com/inspektor-gadget/inspektor-gadget/blob/02eeb75d20cea7e432c585b7a8fc6f280ffb2e04/pkg/operators/oci-handler/oci.go#L65

https://github.com/inspektor-gadget/inspektor-gadget/blob/551966b6cf65505d4af32db2e699c92b73f2894e/pkg/operators/kubemanager/kubemanager.go#L80

etc 

If two operators/runtimes use the same key, then it's going to cause a name clash as both will try to register a CLI flag with the same name. 

Btw, AFAIU we don't have any mechanism to protect against it, perhaps we can throw an error (or print a warning) instead of silently ignoring it. WDYT @flyth?

Reply:
This one's really tricky from the UX perspective.
I think a warning would be good - but I wouldn't throw an error as the user usually can't fix that "issue".

I think a solution (but probably out of scope here) could be (not all cases covered):
* add a "shared" flag to Params (this would also cover other use cases like `--output`)
* if param already exists and is shared (for both instances) AND shares the same TypeHint, reuse it (operators would share that param); descriptions could be appended/merged
* if a param already exists and is not shared, instead use a fully qualified name (e.g. `--oci.paramabc` or `--oci-paramabc`)
* if a param already exists and the existing one is not shared but the current one is, demote the existing one to fully qualified (see above)

I think for env I would always use the fully qualified ones.

Reply:
> I think for env I would always use the fully qualified ones.

That's a good point, so perhaps it's actually better to have long env variables names and avoid the name clash issue there.

Reply:
Makes sense. I think since we haven't been using envs for CLIs so it would be fine to start with the long names. I see [couple of places where shorter versions](https://github.com/inspektor-gadget/inspektor-gadget/blob/b6b58eb0ab96e43b187075763339d643cb6fc519/pkg/container-utils/containerutils.go#L58) are being used so we might need couple of steps to get rid of them!